reactive/containerd.py

```diff
@@ -538,19 +539,24 @@ def configure_nvidia(reconfigure=True):
     key_urls = config("nvidia_apt_key_urls").split()
     formatted_key_urls = [
         key_url.format(
-            id=os_release_id, version_id=os_release_version_id, version_id_no_dot=os_release_version_id_no_dot
+            id=os_release_id,
+            version_id=os_release_version_id,
+            version_id_no_dot=os_release_version_id_no_dot,
         )
         for key_url in key_urls
     ]
     if formatted_key_urls:
         fetched_keys = fetch_url_text(formatted_key_urls)
         if not all(fetched_keys):
-            status.blocked("Failed to fetch nvidia_apt_key_urls.")
+            set_state("containerd.nvidia.fetch_keys_failed")
```

Comment:
if you create a flag, you should clear it somewhere...

Maybe this? 
```python
    if formatted_key_urls:
        fetched_keys = fetch_url_text(formatted_key_urls)
        if not all(fetched_keys):
            set_state("containerd.nvidia.fetch_keys_failed")
            return
        remove_state("containerd.nvidia.fetch_keys_failed")
```

Is there a reason for this being a reactive flag?  I don't see any other changes where there is a reactive hook run based on this state change -- only a check for set  in update_status.|

Another thing to be aware of is that `install_nvidia_drivers` will continue on in the event of this failure path of `configure_nvidia_sources`.  Perhaps a successful `configure_nvidia_sources` should return True or maybe the `NVIDIA_SOURCS_FILE` which was created -- and this failure path returns None so the caller can react accordingly

Reply:
I had originally seen some cases where something else that happened later on removed the block. It might have been because of an error in my code, but AIUI because the block was `@atexit` that it didn't have to wait for an update_status, but ran at the end of each invocation, and with it being this way it would always be caught by an update_status.